AK/Tests/TestSourceLocation.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2021, Andrew Kaster <andrewdkaster@gmail.com>
+ * Copyright (c) 2021, Brian Gianforcaro <bgianf@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/SourceLocation.h>
+#include <AK/StringView.h>
+#include <AK/TestSuite.h>
+
+TEST_CASE(basic_scenario)
+{
+    auto location = SourceLocation::current();
+    EXPECT_EQ(StringView(__FILE__), location.file_name());
+    EXPECT_EQ(StringView(__FUNCTION__), location.function_name());
+    EXPECT(location.line_number() != 0);
```

Comment:
can we be more specific here and use the exact number?

Reply:
sure